core/core-api/src/main/java/io/fabric8/launcher/core/spi/Application.java

```diff
@@ -23,8 +25,19 @@
 public @interface Application {
 
     enum ApplicationType {
+
         LAUNCHER,
-        OSIO
+        OSIO;
+
+        public static ApplicationType fromHeader(HttpServletRequest request) {
+            // If X-App is not specified, assume fabric8-launcher
+            final String app = Objects.toString(request.getHeader("X-App"), "launcher").toUpperCase();
+            try {
+                return valueOf(app);
+            } catch (IllegalArgumentException iae) {
+                throw new IllegalArgumentException("Header 'X-App' has an invalid value: " + app);
```

Comment:
should we maybe use the value iae so we don't swallow it?

Reply:
```suggestion
                new IllegalArgumentException("Header 'X-App' has an invalid value: " + app, iae);
```

would make sense?